linux/LinuxProcessList.c

```diff
@@ -404,6 +404,84 @@ static void LinuxProcessList_readIoFile(LinuxProcess* process, const char* dirna
 
 
 
+typedef struct LibraryData {
+    uint64_t size;
+    bool exec;
+} LibraryData;
+
+static void LinuxProcessList_calcLibSize_helper(
+   ATTR_UNUSED int key, void* value, void* data
+) {
+   if(!data)
+      return;
+
+   if(!value)
+      return;
+
+   LibraryData * v = (LibraryData *)value;
+   uint64_t * d = (uint64_t *)data;
+   if(!v->exec)
+      return;
+
+   *d += v->size;
+}
+
+static uint64_t LinuxProcessList_calcLibSize(const char* dirname, const char* name) {
+   char filename[MAX_NAME+1];
+   xSnprintf(filename, sizeof(filename), "%s/%s/maps", dirname, name);
+   FILE* mapsfile = fopen(filename, "r");
+   if (!mapsfile)
+      return 0;
+
+   Hashtable * ht = Hashtable_new(64, true);
+   if(!ht) {
+      fclose(mapsfile);
+      return 0;
+   }
+
+   char buffer[1024];
+   while(fgets(buffer, sizeof(buffer), mapsfile)) {
+      uint64_t map_start;
+      uint64_t map_end;
+      char map_perm[16];
+      uint32_t map_dummy_1;
+      int map_dummy_2;
+      int map_dummy_3;
+      uint32_t map_inode;
```

Comment:
Pretty sure this is a 64bit number on 64bit platforms, so 'uint64_t' here for the win.

(looks like the kernel uses "seq_put_decimal_ull(m, " ", ino);" over in fs/proc/task_mmu.c for example)